frontend/src/individual_edit.tsx

```diff
@@ -244,6 +250,98 @@ export function IndividualEdit({ id }: { id: string | undefined }) {
     individual && setIndividual({ ...individual, [field]: value });
   };
 
+  const isDateOkay = (date: string) => {
+    const recordDate = new Date(date).getTime();
+    const today = new Date().getTime();
+    if (today - recordDate < 0) {
+      userMessage("Ange ett datum som inte ligger i framtiden.", "warning");
+      return false;
+    } else {
+      return true;
+    }
+  };
+
+  const handleNewWeight = () => {
+    if (!weightDate) {
+      userMessage("Ange ett datum för viktmätningen", "warning");
+      return;
+    }
+    if (weight <= 0) {
+      userMessage("Ange en vikt större än 0.", "warning");
+      return;
+    }
+    if (!isDateOkay(weightDate)) {
+      return;
+    }
+
+    // update the local individual state
+    updateField("weights", [
+      ...individual?.weights,
+      { date: weightDate, weight: weight },
+    ]);
+
+    const newWeightRecord = {
+      number: individual?.number,
+      herd: individual?.herd,
+      weights: [...individual?.weights, { date: weightDate, weight: weight }],
+    };
+
+    // send weight record to the backend
+    patch("/api/individual", newWeightRecord)
+      .then((json) => {
+        if (json.status == "success") {
+          userMessage("Viktmätningen har lagts till.", "success");
+          return;
+        } else {
+          userMessage("Något gick fel.", "error");
+        }
+      })
+      .then(() => {
+        setWeight(0);
+        setWeightDate(null);
+      });
+  };
```

Comment:
I'm not sure it needs addressing, but for my understanding; this doesn't handle errors if the request fails, right (e.g. browser has lost internet connection)?

Reply:
Right, I forgot the `catch`. Added it now, I think that should solve it.